### DISCUSSION TITLE
Dark Lord - white to golden text

### DISCUSSION THREAD

**Discussion by oli25** [diasurgical/DevilutionX#2112] [Ideas]
It's super weird that last boss has white text color. Can we change it to golden?
If it's not going to make it into DevilutionX can somebody please tell me how to code it?

**Comment by julealgon** [comment]
The color depends on the monster being classified as a "boss", and Diablo is not. So the question here is not specifically about the color of the text, but whether or not we want to "convert" diablo into a proper super boss unit (like Butcher, Leoric, etc).

My vote is yes.

> If it's not going to make it into DevilutionX can somebody please tell me how to code it?

I suspect this would have good acceptance, but the change should be easy: search for the flag that indicates whether an enemy is a unique boss, and turn it "on" for Diablo. The gold text color should follow naturally.

**Reply by qndel** [reply]
turn it on and then his items will get artificially boosted = you changed the game

**Reply by agris-codes** [reply]
Diablo doesn’t drop items, killing him triggers end game movie. He doesn’t even have corpse frames.

**Reply by AJenbo** [reply]
he does drop items

**Reply by oli25** [reply]
and replacing basic diablo with added diablo boss requires dun editing?

**Reply by qndel** [reply]
no

**Reply by agris-codes** [reply]
> 
> 
> he does drop items

Is this a hellfire-specific thing, or a recent change in the devilutionX build? I killed him in a 1.2.1 diablo game and instantly got the end movie.

**Reply by AJenbo** [reply]
no, he drops and item bleeds for a few sec and then the movie starts. To pick it up you must save and load first.

**Reply by ChaosMarc** [reply]
the save and reload mechanism doesn't work anymore on the latest master build. I don't know since when, but it's been at least a few weeks (it still works in 1.4.1). currently tested on d055aa5. It's been my first drop ever from Diablo himself so I wanted to grab the item but I'm not able to :(
![Screenshot-from-2023-05-23-10-54-06](https://github.com/diasurgical/devilutionX/assets/1343751/effc557d-c476-41ab-8733-78db81adb230)

[single_0.zip](https://github.com/diasurgical/devilutionX/files/11541786/single_0.zip)

Am I doing something wrong or should I open a bug ticket?

**Reply by DakkJaniels** [reply]
> the save and reload mechanism doesn't work anymore on the latest master build. I don't know since when, but it's been at least a few weeks (it still works in 1.4.1). currently tested on [d055aa5](https://github.com/diasurgical/devilutionX/commit/d055aa5cbd1439edcc521df7b299edd9edeac19c). It's been my first drop ever from Diablo himself so I wanted to grab the item but I'm not able to :( 
> 
> [single_0.zip](https://github.com/diasurgical/devilutionX/files/11541786/single_0.zip)
> 
> Am I doing something wrong or should I open a bug ticket?

Why is this in this Discussion?

The bug that allowed you to move around after loading a save where Diablo is dead was fixed in #4149

**Reply by ChaosMarc** [reply]
> Why is this in this Discussion?

Because I was searching for related issues/discussions and realized to late how old this one was 🤦 should we move this to a new discussion?

> The bug that allowed you to move around after loading a save where Diablo is dead was fixed in #4149

But why should picking up items be prevented if Diablo is able to drop items? I was following the discussion in the issue you linked back then and would have argued against preventing walking if I had realized that preventing any action was the fix for the camera shaking. I know this is a very controversal topic but I don't think this is the best solution and would like to argue for some other way

**Reply by DakkJaniels** [reply]
> But why should picking up items be prevented if Diablo is able to drop items? I was following the discussion in the issue you linked back then and would have argued against preventing walking if I had realized that preventing any action was the fix for the camera shaking. I know this is a very controversal topic but I don't think this is the best solution and would like to argue for some other way

Because it was obviously a bug. The state wasn't restored correctly upon loading a game and that was fixed. DevilutionX has fixed plenty of other bugs that had useful abilities/features, like fire arrows stopping monster healing, or getting an extra frame of hit recovery by wearing a fast, faster, and fastest hit recovery item. If you want to get the item, you can always save and load your file in version 1.4.1

**Reply by StephenCWills** [reply]
It's not that picking up items should be prevented. That's just a consequence of how Blizzard chose to implement the transition to the ending cinematic. And Save/Load was a workaround found by players, not a feature provided by Blizzard. For instance, it was never even an option in Multiplayer.

One potential solution would be an option to disable the ending cinematic. However, it turns out this is tricky to implement for Multiplayer.

**Reply by ChaosMarc** [reply]
An option to disable the cinematic would be awesome. Wouldn't it be possible to add such an option for SP first and create an open issue for adding MP compatibility later? I have the vague feeling of having read a discussion about this somewhere but can't find the respective issue.

**Reply by StephenCWills** [reply]
> Wouldn't it be possible to add such an option for SP first and create an open issue for adding MP compatibility later?

Yes, it would be possible. But I'm not going to do that.

**Reply by ChaosMarc** [reply]
> Yes, it would be possible. But I'm not going to do that.

Fair enough 😅

Seems like a manageable thing for a first PR. Maybe if I can find some time 🙄

**Reply by AJenbo** [reply]
a fix would be to not have him drop anything

**Reply by Korth999** [reply]
Diablo already drops trash, like any other "normal" monster. Some of the time.
There's no great need to "fix" anything since nothing he could drop is really worth sticking around to collect.

**Reply by ChaosMarc** [reply]
It's not so much HIS drop but the drops of the monsters surrounding him. In vanilla you always have to take care of collecting everything on the ground before killing him because afterwards you can't pick it up. I often had magical items lying around after clearing his "box"

**Reply by AJenbo** [reply]
How can you even manage to do that in the few sec before the game ends :o

**Reply by ChaosMarc** [reply]
sometimes the time suffices but most of the times when I saw an item too late and had to reload and do the fight again just to collect the item up front... IMHO this is horrible game design. 

btw. I now have a [fork](https://github.com/ChaosMarc/devilutionX/commit/494ad8f35eccf21262b7adb20e3b3e0026d531b6) in which I added an option to stay in the game after killing diablo. It works as expected and I can't find any problems in SP nor MP. The only thing thats bothering me still is the fact das Diablos dying animation is playing indefinitely. Not sure if this would be an acceptable fix or if something other has to be found but I fear I would need help for anything more sophisticated 😇

**Reply by DakkJaniels** [reply]
It's not an acceptable fix, if it was, it could have been done at any time (I'm sure others have implemented it before me, but Middle Earth mod had that in over a year ago, I think).
